vcs/versioning.go

```diff
@@ -131,33 +131,47 @@ func GetUncommittedChanges(vcs Vcs) ([]FileChange, error) {
 }
 
 func GetFileHistory(vcs Vcs, file string, clock Clock) (*FileHistory, error) {
-	output, err := vcs.Log("--format=%at", "--", file)
+	output, err := vcs.Log("--follow", "--name-status", "--format=%at", "--", file)
+	if err != nil {
+		return nil, err
+	}
+	timestamps, err := getCommitTimestamps(file, output)
 	if err != nil {
 		return nil, err
 	}
-	lines := Split(output, "\n")
-	lines = lines[0 : len(lines)-1]
-	lineCount := len(lines)
 	defaultYear := clock.Now().Year()
 	history := FileHistory{
 		CreationYear:    defaultYear,
 		LastEditionYear: defaultYear,
 	}
-	if lineCount > 0 {
-		timestamp, err := strconv.ParseInt(lines[lineCount-1], 10, 64)
-		if err != nil {
-			return nil, err
-		}
-		history.CreationYear = time.Unix(timestamp, 0).Year()
+
+	if len(timestamps) > 0 {
+		minTimestamp := timestamps[len(timestamps)-1]
+		maxTimestamp := timestamps[0]
+		history.CreationYear = time.Unix(minTimestamp, 0).Year()
+		history.LastEditionYear = time.Unix(maxTimestamp, 0).Year()
+
 	}
-	if lineCount > 1 {
-		timestamp, err := strconv.ParseInt(lines[0], 10, 64)
+	return &history, nil
+}
+
+func getCommitTimestamps(file string, log string) ([]int64, error) {
+	var result []int64
+	lines := Split(Replace(log, "\n\n", "\n", -1), "\n")
+	lines = lines[0 : len(lines)-1]
+	for i := 1; i < len(lines); i+=2 {
+		line := lines[i]
+		if Split(line, "\t")[0] == "R100" {
```

Comment:
Please define a constant for "R100" with name such as `renameSameContent`.

What about handling "C100" similarly? Probably much rarer though.

Reply:
Fixed in e61ffcf40a388376dfb457b69c7b63ab5dc29793